vsintegration/tests/UnitTests/CompletionProviderTests.fs

```diff
@@ -583,6 +583,14 @@ type T() =
 """
     VerifyNoCompletionList(fileContents, "member this.M(p:int, h")
 
+[<Test>]
+let ``Completion list on abstract member type signature contains modules and types but not keywords or functions`` =
+    let fileContents = """
+type Interface =
+    abstract member Eat: l
```

Comment:
One problem here is that an arbitrary identifer is valid here

```
type Interface =
    abstract member Eat: nameOfParameter: int -> int
```

Does this work correctly, in particular is there electric completion when only one autocomplete suggestion remains or can the user continue to type the name of the identifier?

Reply:
The only difference in that regard is that the user will see filtered completions (as they are typing the line, we can't tell in advance if what we're seeing is meant to be an identifier or a type).

If you type a semicolon or a space (the picture below is what I think you're asking), the thing does not get autocompleted. You can also keep typing past the last thing on the list of course. The PR does not affect that behavior at all.

![devenv_BajE9Z61VN](https://user-images.githubusercontent.com/5063478/167306689-1118ea19-0f57-4431-932e-58c6fbb8d801.png)

If you go back and change the identifier, you'll unfortunately get unfiltered completions (with and without my changes). This is because I've only added a trigger for `VisitType`. `SyntaxVisitorBase` does not have an overridable member for traversing abstract members, so I don't have much to lean on in `TryGetCompletionContext`.

![devenv_531neq1Z8q](https://user-images.githubusercontent.com/5063478/167306949-487aec70-8664-4776-b251-87a1c7017bcb.png)